src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -46,9 +47,38 @@ public function create(string $resourceClass, string $property, array $options =
     {
         $propertyMetadata = $this->decorated->create($resourceClass, $property, $options);
 
+        list($normalizationGroups, $denormalizationGroups) = $this->getGroups($options, $resourceClass);
+
+        $serializerClassMetadata = $this->serializerClassMetadataFactory->getMetadataFor($resourceClass);
+        $groups = [];
+        foreach ($serializerClassMetadata->getAttributesMetadata() as $serializerAttributeMetadata) {
+            if ($property === $serializerAttributeMetadata->getName()) {
+                $groups = $serializerAttributeMetadata->getGroups();
+                break;
+            }
+        }
+
+        /*
+         * Set readable/writable based on matching normalization/denormalization groups,
+         * but never reset a false value as it could be unreadable/unwritable for other reasons
+         *
+         * If normalization/denormalization groups are not specified, the property is implicitly readable/writable
+         */
+        if (false !== $propertyMetadata->isReadable()) {
```

Comment:
coud L67-75 be moved into a private `::getPropertyMetadata()` method? Then those comments could go in the method PHPDoc. That would increase readability IMO